File: ebs/src/main/kotlin/com/bft/ebs/services/impl/AuthorService.kt
package com.bft.ebs.services.impl

import com.bft.ebs.dto.AuthorDto
import com.bft.ebs.repositories.impl.AuthorRepository
import com.bft.ebs.services.EbsMapper
import com.bft.ebs.services.interfaces.AuthorService
import org.springframework.stereotype.Service

@Service
class AuthorService(private val authorRepository: AuthorRepository) :
    AuthorService {
    override fun getAuthor(id: Long): AuthorDto {
        return EbsMapper.mapAuthorToAuthorDto(authorRepository.read(id))
    }

    override fun getAllAuthors(): List<AuthorDto> {
        return authorRepository.findAll().map {
            EbsMapper.mapAuthorToAuthorDto(it)
        }
    }

    override fun createAuthor(authorDto: AuthorDto): Long {
        return authorRepository.create(EbsMapper.mapAuthorDtoToAuthor(authorDto)).id
            ?: throw IllegalStateException("Couldn't create author.")
    }

    override fun deleteAuthor(id: Long): Boolean {
        return authorRepository.delete(id)
    }

    override fun updateAuthor(authorDto: AuthorDto): Boolean {
        return authorRepository.update(
            EbsMapper.mapAuthorDtoToAuthor(authorDto)
        )
    }
}

File: front/ebs-frontend/src/services/AuthorService.js
import axios from "axios";
import { REST_API_BASE_URL } from '../../config'

const SERVICE_PATH = REST_API_BASE_URL + 'authors/'

export const listAuthors = () => axios.get(SERVICE_PATH + 'all');

export const createAuthor = (author) => axios.post(SERVICE_PATH + 'create', author)

export const getAuthor = (authorId) => axios.get(SERVICE_PATH + 'get/' + authorId)

export const updateAuthor = (authorId, author) => axios.put(SERVICE_PATH + 'update/' + authorId, author)

export const deleteAuthor = (authorId) => axios.delete(SERVICE_PATH + 'delete/' + authorId)